File: Sources/ApprovalView.swift
import SwiftUI

/// The approval sheet is intentionally light even over the dark call —
/// per the locked artboard: sheet · one decision · clear code.
struct ApprovalView: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Capsule()
                .fill(DT.inkLight.opacity(0.16))
                .frame(width: 38, height: 5)
                .frame(maxWidth: .infinity)
                .padding(.top, 10)

            Text("Approve change?")
                .font(.system(size: 26, weight: .heavy))
                .kerning(-0.8)
                .foregroundStyle(DT.inkLight)
                .padding(.top, 18)

            HStack(spacing: 5) {
                Text("Cline wants to edit").foregroundStyle(DT.inkLight.opacity(0.55))
                Text("auth.ts")
                    .font(.system(size: 12.5, design: .monospaced))
                    .foregroundStyle(DT.inkLight)
                    .padding(.horizontal, 6).padding(.vertical, 2)
                    .background(DT.surface2Light)
                    .clipShape(RoundedRectangle(cornerRadius: 5, style: .continuous))
            }
            .font(.system(size: 14))
            .padding(.top, 6)

            VStack(alignment: .leading, spacing: 5) {
                Text("+ export function requireAuth()")
                Text("+   verifyJwt(req)")
                Text("+   next()")
            }
            .font(.system(size: 12.5, design: .monospaced))
            .foregroundStyle(DT.diffGreenOnLight)
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, 15).padding(.vertical, 13)
            .background(DT.surface2Light)
            .clipShape(RoundedRectangle(cornerRadius: DT.rControl, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: DT.rControl, style: .continuous)
                .strokeBorder(DT.inkLight.opacity(0.08), lineWidth: 0.8))
            .padding(.top, 16)

            HStack(spacing: 14) {
                HStack(spacing: 4) {
                    Text("+12").foregroundStyle(DT.diffGreenOnLight)
                    Text("−3").foregroundStyle(DT.danger)
                }
                Text("auth.ts")
                Text("branch drive/auth")
            }
            .font(.system(size: 11, design: .monospaced))
            .foregroundStyle(DT.inkLight.opacity(0.35))
            .padding(.top, 10)

            HStack(spacing: 10) {
                Button { store.denyEdit() } label: {
                    Text("Deny")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundStyle(DT.inkLight)
                        .frame(maxWidth: .infinity)
                        .frame(height: 52)
                        .background(DT.surface2Light)
                        .clipShape(RoundedRectangle(cornerRadius: DT.rHero, style: .continuous))
                }
                .buttonStyle(Pressable())

                Button { store.allowEdit() } label: {
                    Text("Allow")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundStyle(.white)
                        .frame(maxWidth: .infinity)
                        .frame(height: 52)
                        .background(DT.heroGradient)
                        .clipShape(RoundedRectangle(cornerRadius: DT.rHero, style: .continuous))
                        .shadow(color: DT.violet.opacity(0.35), radius: 12, y: 6)
                }
                .buttonStyle(Pressable())
                .frame(maxWidth: .infinity)
            }
            .padding(.top, 20)

            Text("Nothing lands without you — every edit is yours to allow.")
                .font(.system(size: 11))
                .foregroundStyle(DT.inkLight.opacity(0.35))
                .frame(maxWidth: .infinity)
                .padding(.top, 14)

            Spacer(minLength: 0)
        }
        .padding(.horizontal, 20)
        .background(Color.white.ignoresSafeArea())
        .environment(\.colorScheme, .light)
    }
}

File: web/src/views/home/ApprovalView.js
// ApprovalView.swift — the approval sheet is intentionally light even over the
// dark call: sheet · one decision · clear code. app.js presents it as a medium,
// light sheet when `store.showApproval` flips; Deny/Allow decide and dismiss.
import { html, useEffect, useObservable } from "../../ui.js";
import { DiffLine } from "../../components.js";
import { ctx } from "./shared.js";

const LINES = ["+ export function requireAuth()", "+   verifyJwt(req)", "+   next()"];

export function ApprovalView() {
  const s = useObservable(ctx.store);
  const nav = ctx.nav;

  // A swipe-down mirrors `.sheet(isPresented:)` — the binding goes false.
  useEffect(() => () => { if (ctx.store.showApproval) ctx.store.set({ showApproval: false }); }, []);

  const decide = (allow) => {
    if (allow) s.allowEdit(); else s.denyEdit();
    nav.dismiss();
  };

  return html`<div class="ap" data-surface="approval" role="document" aria-labelledby="ap-title">
    <h2 id="ap-title">Approve change?</h2>
    <div class="who">Cline wants to edit <span class="file">auth.ts</span></div>
    <div class="diff" role="group" aria-label="Proposed change, 3 added lines">
      ${LINES.map((t) => html`<${DiffLine} key=${t} text=${t} added />`)}
    </div>
    <div class="stats" aria-label="12 additions, 3 deletions, auth.ts, branch drive/auth">
      <span><span class="plus">+12</span> <span class="minus">−3</span></span>
      <span>auth.ts</span>
      <span>branch drive/auth</span>
    </div>
    <div class="btns">
      <button class="deny" onClick=${() => decide(false)}>Deny</button>
      <button class="allow" onClick=${() => decide(true)}>Allow</button>
    </div>
    <div class="foot">Nothing lands without you — every edit is yours to allow.</div>
  </div>`;
}
